src/mobile/src/main/java/tupperdate/android/ui/layout/SwipeStack.kt

```diff
@@ -0,0 +1,252 @@
+    initialValue: SwipeStackValue,
+    clock: AnimationClockObservable,
+    confirmStateChange: (SwipeStackValue) -> Boolean = { true },
+) : SwipeableState<SwipeStackValue>(
+    initialValue = initialValue,
+    clock = clock,
+    animationSpec = AnimationSpec,
+    confirmStateChange = confirmStateChange,
+) {
+    /**
+     * Whether the swipe stack is swiped.
+     */
+    val isSwiped: Boolean
+        get() = value != SwipeStackValue.NotSwiped
+
+    /**
+     * Swipe the stack to the end with an animation.
+     */
+    fun swipeEnd() {
+        animateTo(SwipeStackValue.SwipedEnd)
+    }
+
+    /**
+     * Swipe the stack to the start with an animation.
+     */
+    fun swipeStart() {
+        animateTo(SwipeStackValue.SwipedStart)
+    }
+}
+
+@ExperimentalMaterialApi
+private fun SwipeStackState.angle(): Float {
+    val progress = progress
+    val angle: (SwipeStackValue) -> Float = {
+        when (it) {
+            SwipeStackValue.NotSwiped -> 0f
+            SwipeStackValue.SwipedEnd -> SwipeStackMaxAngle
+            SwipeStackValue.SwipedStart -> -SwipeStackMaxAngle
+        }
+    }
+    val delta = angle(progress.to) - angle(progress.from)
+    return angle(progress.from) + (delta * progress.fraction)
+}
+
+/**
+ * Create and remember a [SwipeStackState] with the default animation clock.
+ *
+ * @param initialValue the initial value of the state.
+ * @param confirmStateChange optional callback invoked to confirm or veto a pending state change.
+ */
+@ExperimentalMaterialApi
+@Composable
+fun rememberSwipeStackState(
+    initialValue: SwipeStackValue = SwipeStackValue.NotSwiped,
+    confirmStateChange: (SwipeStackValue) -> Boolean = { true },
+): SwipeStackState {
+    val clock = AnimationClockAmbient.current.asDisposableClock()
+    // TODO : Remember this across configuration changes, with rememberSavedInstanceState.
+    return remember {
+        SwipeStackState(initialValue, clock, confirmStateChange)
+    }
+}
+
+/**
+ * A swipe stack displays multiple swipeable items, such that they can be dismissed in both start
+ * and end directions.
+ *
+ * @param items the items that are displayed in the stack. These items are keyed, and you must
+ *              therefore make sure that they are all unique (for instance with a unique id).
+ * @param modifier the modifier for this composable.
+ * @param padding the amount of padding between the stacked items.
+ * @param peekCount the count of items that are peeking behind the front item.
+ * @param swipeStackState the [SwipeStackState] that is used for this composable.
+ * @param content a lambda that generates the body to display for each item.
+ */
+@ExperimentalMaterialApi
+@Composable
+fun <T> SwipeStack(
+    items: List<T>,
+    modifier: Modifier = Modifier,
+    padding: Dp = SwipeStackDefaultPadding,
+    peekCount: Int = SwipeStackDefaultOverlapCount,
+    swipeStackState: SwipeStackState = rememberSwipeStackState(),
+    content: @Composable (T) -> Unit,
```

Comment:
I think this (T) parameter could be annoying, because all of our composables don't have it.

Reply:
The `T` is an item from the list of stacked elements. It's used to build the associated content composable.

Please look at how lists work in Compose; it's exactly the same pattern.

Reply:
(here's an [example](https://cs.android.com/androidx/platform/frameworks/support/+/androidx-master-dev:compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/lazy/LazyFor.kt;l=54) of the said pattern)